server/studyspaces.py

```diff
@@ -251,21 +248,37 @@ def book_room():
     try:
         user = User.get_user()
         if user and user.email:
-            user.email = contact["email"]
-            sqldb.session.commit()
+            if "email" in contact:
+                user.email = contact["email"]
+                sqldb.session.commit()
+            else:
+                contact["email"] = user.email
         user = user.id
     except ValueError:
         user = None
 
-    resp = studyspaces.book_room(room, start.isoformat(), end.isoformat(), **contact)
-    if "results" in resp:
+    if lid == 1:
+        sessionid = get_wharton_sessionid()
+        if not sessionid:
+            return jsonify({"results": False, "error": "You must pass a sessionid when booking a Wharton GSR!"})
+        resp = wharton.book_reservation(sessionid, room, start, end)
+        save_wharton_sessionid()
```

Comment:
We should only save the session id if the booking goes through. That way we know it's valid.